lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
I think we should make Resource._add_global_resource() part of the public API. Then any custom threads (or their containing class) just need to have a free() method to clean themselves up, and register with Resource.
We'd need to document this requirement, of course.
Happy with that approach?